src/main/java/com/spartronics4915/frc2019/subsystems/CargoChute.java

```diff
@@ -95,52 +111,50 @@ public void onLoop(double timestamp)
                 SystemState newState = defaultStateTransfer();
                 switch (mSystemState)
                 {
-                    case RAMPING:
+                    case INTAKING:
                         if (mStateChanged)
                             mRampMotor.set(ControlMode.PercentOutput, Constants.kRampSpeed);
-                        if (ballInPosition() && !isInManual() && newState == mSystemState)
-                            newState = SystemState.HOLDING;
                         break;
                     case HOLDING:
                         if (mStateChanged)
                             mRampMotor.set(ControlMode.PercentOutput, 0.0);
-                        if (!ballInPosition() && !isInManual() && newState == mSystemState)
-                            newState = SystemState.RAMPING;
                         break;
                     case EJECTING:
                         if (mStateChanged)
                             mRampMotor.set(ControlMode.PercentOutput, -Constants.kRampSpeed);
                         break;
-                    case LOWERING:
-                        if (mStateChanged)
-                            mRampSolenoid.set(Constants.kRampSolenoidRetract);
-                        break;
-                    case RAISING:
-                        if (mStateChanged)
-                            mRampSolenoid.set(Constants.kRampSolenoidExtend);
-                        break;
-                    case SHOOTING_ROCKET:
+                    case SHOOTING:
                         if (mStateChanged)
                         {
-                            mRampSolenoid.set(Constants.kRampSolenoidRetract);
                             mCargoTimer.start();
-                            mRampMotor.set(ControlMode.PercentOutput, Constants.kRampSpeed);
+                            if (mWantedState == WantedState.SHOOT_BAY)
+                                mRampSolenoid.set(Constants.kRampSolenoidExtend);
```

Comment:
You can't assume that the motor won't be running here, so you should set its percent demand to 0.